web/packages/teleterm/src/ui/DocumentAccessRequests/NewRequest/NewRequest.tsx

```diff
@@ -72,22 +90,28 @@ export function NewRequest() {
     updateResourceKind,
     prevPage,
     requestableRoles,
-    isLeafCluster,
     nextPage,
     agents,
     addedItemsCount,
-  } = useNewRequest();
+  } = useNewRequest(props.rootCluster);
+  const { documentsService, localClusterUri } = useWorkspaceContext();
 
   const requestStarted = addedItemsCount > 0;
 
-  // Leaf clusters do not allow role requests, so we do not show that option in the UI if leaf
-  const filteredAgentOptions = useMemo(
-    () =>
-      agentOptions.filter(agent =>
-        isLeafCluster ? agent.value !== 'role' : agent
```

Comment:
I removed this check because for leaf clusters we would have a blank 'Access Requests' screen when the resources can requested from the main view. 
Also, the roles apply to the entire cluster, so I believe they can always be displayed.